lib/practice_exercises.rb

```diff
@@ -1,13 +1,60 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(1)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+
+  alphanumeric_char = string.downcase.gsub(/[^a-z0-9]/i, '')
+
+  if string == nil
+    return true
+  elsif alphanumeric_char == reverse_string(alphanumeric_char)
+    return true
+  else
+    return false
+  end
+end
+
+# helper method to reverse the string
+# Time Complexity: O(n)
+def reverse_string(string)
+  str_arr = string.chars
+  i = 0
+  j = str_arr.length - 1
+
+  while i < j
+    temp = str_arr[i]
+    str_arr[i] = str_arr[j]
+    str_arr[j] = temp
+    i += 1
+    j -= 1
+  end
+  return str_arr.join
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+
+# Time Complexity: O(n2) => we have nested loop
+# Space Complexity: O(1)
+# split the arr
 def longest_prefix(strings)
```

Comment:
👍 However you loop through the list of strings, n, and loop through each character in the 1st string, m.  So the time complexity is O(nm).

Similarly you are building a prefix so the space complexity is O(m).